utils/utils.py

```diff
@@ -301,10 +301,10 @@ def get_latest_run_id(log_path: str, env_id: str) -> int:
     """
     max_run_id = 0
     for path in glob.glob(os.path.join(log_path, env_id + "_[0-9]*")):
-        file_name = os.path.basename(path)
-        ext = file_name.split("_")[-1]
-        if env_id == "_".join(file_name.split("_")[:-1]) and ext.isdigit() and int(ext) > max_run_id:
-            max_run_id = int(ext)
+        run_id = path.split("_")[-1]
+        path_without_run_id = path[:-len(run_id)-1]
+        if path_without_run_id.endswith(env_id) and run_id.isdigit() and int(run_id) > max_run_id:
```

Comment:
I think those checks here are redundant due to the structure of the glob regex above.

This whole function could be simplified to:

```python
return max(int(path.split("_")[-1] for path in glob.glob(os.path.join(log_path, env_id + "_[0-9]*")))
```